src/Component/src/Translation/Provider/ImmutableTranslationLocaleProvider.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Translation\Provider;
+
+final class ImmutableTranslationLocaleProvider implements TranslationLocaleProviderInterface
+{
+    /** @var array */
+    private $definedLocalesCodes;
+
+    private string $defaultLocaleCode;
+
+    public function __construct(array $definedLocalesCodes, string $defaultLocaleCode)
+    {
+        $this->definedLocalesCodes = $definedLocalesCodes;
+        $this->defaultLocaleCode = $defaultLocaleCode;
+    }
```

Comment:
```suggestion
    public function __construct(
        private array $definedLocalesCodes,
        private string $defaultLocaleCode,
     ) {
    }
```

Reply:
Hum should not be a bc-break indeed cause definedLocalesCodes is private an is an array on contructor arguements.